dot/parachain/dispute/types/dispute.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"bytes"
+
+	parachainTypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// Comparator for ordering of disputes.
+type Comparator struct {
+	SessionIndex  parachainTypes.SessionIndex `scale:"1"`
+	CandidateHash common.Hash                 `scale:"2"`
+}
+
+// Dispute is a dispute for a candidate.
+// It is used as an item in the btree.BTree ordered by the Comparator.
+type Dispute struct {
+	Comparator    Comparator    `scale:"1"`
+	DisputeStatus DisputeStatus `scale:"2"`
+}
+
+// NewDispute creates a new dispute for a candidate.
+func NewDispute() (*Dispute, error) {
+	disputeStatus, err := NewDisputeStatus()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Dispute{
+		Comparator:    Comparator{},
+		DisputeStatus: disputeStatus,
+	}, nil
+}
+
+// DisputeComparator compares two disputes.
+func DisputeComparator(a, b any) bool {
```

Comment:
if this is just for Dispute types, do inputs need to be generic?

Reply:
This is being used as the `less` argument for btree. It expects this signature.